v3/src/components/graph/utilities/pixi-transition.ts

```diff
@@ -10,109 +10,73 @@ const defaultInterpolation = smoother
+  static anyTransitionActive(targetProps: TransitionPropMap) {
+    const transitionProps = Object.keys(targetProps) as TransitionProp[]
+    for (const propName of transitionProps) {
+      if (targetProps[propName]?.size) {
+        return true
+      }
     }
+    return false
+  }
 
-    let time = 0
-    const duration = this.duration
-    const startTime = performance.now()
-
-    const transitionFrame = () => {
-      const timeRatio = Math.min(1, time / duration)
-      const factor = defaultInterpolation(timeRatio)
-
-      transitionProps.forEach(propKey => {
-        this.xyTransition(propKey, factor)
-      })
+  static transitionStep(targetProps: TransitionPropMap, startProps: TransitionPropMap) {
+    const now = performance.now()
+    const finishedTransitions = new Set<PixiTransition>()
+    const transitionProps = Object.keys(targetProps) as TransitionProp[]
 
-      if (time < duration) {
-        this.frameId = requestAnimationFrame(transitionFrame)
-        time = performance.now() - startTime
-      } else {
-        this.frameId = undefined
-        this.handleOnEnd()
+    transitionProps.forEach(propKey => {
+      const targetProp = targetProps[propKey]
+      const startProp = startProps[propKey]
+      if (!targetProp || !startProp) {
+        return
       }
-    }
 
-    transitionFrame()
-  }
-
-  handleOnEnd() {
-    this.onEndCallback?.()
-    this.isFinished = true
-  }
+      for (const [point, target] of targetProp.entries()) {
+        const transition = target.transition
+        const factor = transition.getFactor(now)
+        if (!point.destroyed) {
+          const start = startProp.get(point) as TransitionTarget
+          const newX = start.x + factor * (target.x - start.x)
+          const newY = start.y + factor * (target.y - start.y)
+          point[propKey].set(newX, newY)
+        }
+        if (factor === 1 || point.destroyed) {
+          targetProp.delete(point)
+          startProp.delete(point)
+          finishedTransitions.add(transition)
+        }
+      }
+    })
 
-  onEnd(callback: () => void) {
-    this.onEndCallback = callback
-    if (this.isFinished) {
-      this.onEndCallback?.()
+    for (const transition of finishedTransitions) {
+      transition.handleOnEnd()
     }
   }
 
-  destroy() {
-    if (this.frameId) {
-      cancelAnimationFrame(this.frameId)
-    }
+  constructor(duration: number, onEndCallback?: () => void) {
+    this.duration = duration
+    this.onEndCallback = onEndCallback
   }
 
-  setTargetXyProp(propKey: SupportedPropKey, point: PIXI.Sprite, x: number, y: number) {
-    let targetProp = this.targetProp[propKey]
-    let startProp = this.startProp[propKey]
-    if (!targetProp || !startProp) {
-      targetProp = this.targetProp[propKey] = new Map()
-      startProp = this.startProp[propKey] = new Map()
-    }
-    targetProp.set(point, { x, y })
-    startProp.set(point, { x: point[propKey].x, y: point[propKey].y })
+  getFactor(now: number) {
```

Comment:
Perhaps `getProgress`? `getFactor` seems sufficiently generic as to be unhelpful.

Reply:
Thanks, I changed it in https://github.com/concord-consortium/codap/pull/1060/commits/225f14e744826088a1a4009ea5f6ddc9d134a54c.